common/scala/build.gradle

```diff
@@ -32,3 +32,7 @@ dependencies {
     compile 'org.apache.kafka:kafka-clients:0.10.0.0'
     compile 'org.apache.httpcomponents:httpclient:4.4.1'
 }
+
+tasks.withType(ScalaCompile) {
+    scalaCompileOptions.additionalParameters = ['-feature', '-deprecation', '-Xfatal-warnings', '-Ywarn-unused-import']
```

Comment:
My 2¢:
  * add `-unchecked`
  * we use `postfixOps` so much that I would add `-language:postfixOps`, or the imports are going to get annoying very fast